pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -102,8 +103,14 @@ type Reconciler struct {
 	timeoutHandler    *reconciler.TimeoutSet
 }
 
-// Check that our Reconciler implements controller.Reconciler
-var _ controller.Reconciler = (*Reconciler)(nil)
+var (
+	// Check that our Reconciler implements controller.Reconciler
+	_ controller.Reconciler = (*Reconciler)(nil)
+
+	// In case of reconcile errors, we store the error in a multierror, attempt
+	// to update, and return the original error combined with any update error
+	merr error
```

Comment:
i find that global variables tend to cause problems (http://wiki.c2.com/?GlobalVariablesAreBad seems like an okay source of info about this) - for example if more than one `Reconcile` happens at a time (and I think it might, I'm not sure) then multiple goroutines could try to write to the simultaneously

Reply:
Thanks for catching this and for the pointers - I've got some reading and investigating to do.
I think it's reasonable to assume that more than one `Reconcile` could happen at the time, so I should fix this.

Reply:
Done